pkg/pillar/types/edgenodetypes.go

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+// EdgeNodeInfo - edge node info from controller
+type EdgeNodeInfo struct {
+	DeviceName     string
+	DeviceID       string
```

Comment:
For the ones that are UUIDs I think it would make sense to define the type as uuid.UUID.

Reply:
Will do.

Reply:
I figured if I change to uuid.UUID then the API in api/proto/config/devconfig.proto  need to be updated too.
Another issue is device id comes from UUIDandVersion struct where uuid is actually a string. Changing that will certainly break at many places. May be just ignore the type change and rename the variables to reflect the UUID, like DeviceID->DeviceUUID ?

Reply:
No need to change proto file since, there is no protobuf datatype for uuid. Thus zedagent/parseconfig.go would just need to parse the string into a uuid. I think we have code in parseconfig which does that for other uuid strings.

Reply:
Resolved and published new commit.